docs/adr/0018-use-tailwind-with-custom-scss.md

```diff
@@ -0,0 +1,136 @@
+- Reduction of mental overhead and improved development speed.
+- Using a utility-based approach makes sharing components in `fxa-components` straightforward. They can accept a `classes` prop passed in from where the component is composed with the utility classes needed passed in, and any additional styling can be done in an external SCSS file where the component was composed as needed.
+
+Cons:
+
+- Much like Option A this approach would also require our developers to have some knowledge of what's available to them, such as each style's corresponding class name, though this is less of a concern because generally each CSS property is mapped to a class name, and is aided by following size and unit patterns across class names. An IDE extension could be used as well to help in development.
+- Generally, each individual style is applied by a single class, so for a highly-styled element it can lead to a set of long, unwieldy class names.
+  - E.g. Tachyons: `bg-white black-70 ph3 ph5-ns pv5 pv6-ns bt b--black-10`
+  - E.g. Tailwind: `shadow-lg code-white text-sm font-mono subpixel-antialiased bg-gray-800 px-5 pb-6 pt-4 rounded-lg leading-normal overflow-hidden`
+  - E.g. Basscss: `h6 caps bold inline-block py1 color-inherit text-decoration-none hover-underline`
+- Introduces at least one more dependency to our stack. In fact, because some components would be shared and some would be project-specific it's likely that this dependency would exist in multiple areas of the repo.
+
+Out of the utility library options, the team has reached a consensus that Tailwind would be the best option for FxA, which has several pros over other options:
+
+- While most libraries offer some level of configuration, many require working in SASS to override variables and maps, and the resulting output may not always be predictable. Additionally, SASS-level configuration forces us to use SASS variables, and yet in [CSS ADR part 1](https://github.com/mozilla/fxa/blob/master/docs/adr/0015-use-css-variables-and-scss.md), we determined we would like to avoid SASS variables in favor of CSS variables. Tailwind mitigates these inconveniences by offering a root-level JS configuration file which is a better option for organization, clarity, and ease-of-use. Furthermore, Tailwind's configuration options are vast and allow us to set our own incremental spacing, custom font-sizes, determine classname conventions, set custom breakpoints, etc. as we see fit to align with the Settings Redesign visual designs.
+- We can use CSS variables to share stored values with any additional custom SCSS when applicable and for future theming options.
+- Several engineers on our team have prior experience with Tailwind and like it.
+- Tailwind is [significantly more popular](https://www.npmtrends.com/tachyons-vs-tailwindcss-vs-basscss) than other options.
```

Comment:
I would add the specific numbers here, since this link goes to a dynamic webpage. It is conceivable that in the future this may not be a true statement. I might consider something along the lines of:
``` At the time of this decision, Tailwind is significantly more popular (description of data + screenshot). This data was gathered using the following npm package download comparison tool (link to npm trends page)```

Reply:
updated!

Reply:
Thanks!